applications/plugins/CGALPlugin/FindCGAL.cmake

```diff
@@ -1,103 +1,133 @@
+
+if (CGAL_FIND_VERSION)
+  # Check if version found is >= required version
+  if (NOT "${CGAL_VERSION}" VERSION_LESS "${CGAL_FIND_VERSION}")
+    set(CGAL_VERSION_OK TRUE)
+  endif()
+else()
+  # No specific version of CGAL is requested
+  set(CGAL_VERSION_OK TRUE)
 endif()
 
-if ( CGAL_DIR )
+# Add flag to fix bug in CGAL 4.1 for Intel compilers. See
+# https://sympa.inria.fr/sympa/arc/cgal-discuss/2013-01/msg00011.html
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
+  if ("${CGAL_VERSION}" VERSION_GREATER "4.0.2")
+    set(CGAL_DEFINITIONS "-DCGAL_CFG_NO_STATEMENT_EXPRESSIONS")
+  endif()
+endif()
 
-    if ( EXISTS "${CGAL_DIR}/CGALConfig.cmake" )
-        include( "${CGAL_DIR}/CGALConfig.cmake" )
-        set( CGAL_FOUND TRUE )
-    endif()
+# Set variables
+set(CGAL_INCLUDE_DIRS ${CGAL_INCLUDE_DIRS} ${CGAL_3RD_PARTY_INCLUDE_DIRS})
+set(CGAL_LIBRARIES ${CGAL_LIBRARY} ${CGAL_3RD_PARTY_LIBRARIES})
 
+# Add GMP and MPFR libraries if defined by CGAL
+if (GMP_LIBRARIES)
+  set(CGAL_LIBRARIES ${CGAL_LIBRARIES} ${GMP_LIBRARIES})
 endif()
-
-if( NOT CGAL_FOUND)
-    # try to find the paths manually...
-    find_path(CGAL_DIR NAMES include/CGAL/Polyhedron_3.h)
-    find_library(CGAL_LIBRARY NAMES CGAL)
-    find_library(CGAL_ImageIO_LIBRARY NAMES CGAL_ImageIO)
-    if(EXISTS "${CGAL_DIR}/include/CGAL/Polyhedron_3.h" AND EXISTS ${CGAL_LIBRARY})
-        set(CGAL_FOUND TRUE)
-        message(STATUS "CGAL found: ${CGAL_LIBRARY} ${CGAL_DIR}")
-        mark_as_advanced(CGAL_DIR CGAL_LIBRARY)
-
-    endif()
+if (MPFR_LIBRARIES)
+  set(CGAL_LIBRARIES ${CGAL_LIBRARIES} ${MPFR_LIBRARIES})
 endif()
 
-if( NOT CGAL_FOUND)
-    if(CGAL_FIND_REQUIRED)
-        MESSAGE(FATAL_ERROR ${CGAL_DIR_MESSAGE})
-    else()
-        if(NOT CGAL_FIND_QUIETLY)
-            MESSAGE(STATUS ${CGAL_DIR_MESSAGE})
-        endif()
-    endif()
+# Try compiling and running test program
+if (DOLFIN_SKIP_BUILD_TESTS)
+  set(CGAL_TEST_RUNS TRUE)
+elseif (CGAL_INCLUDE_DIRS AND CGAL_LIBRARIES)
+
+  # Set flags for building test program
+  set(CMAKE_REQUIRED_INCLUDES ${CGAL_INCLUDE_DIRS})
+  set(CMAKE_REQUIRED_LIBRARIES ${CGAL_LIBRARIES})
+  set(CMAKE_REQUIRED_FLAGS ${CGAL_CXX_FLAGS_INIT})
+
+  # Add all previusly found Boost libraries - CGAL doesn't appear to supply
+  # all necessary Boost libs (test with Boost 1.50 + CGAL 4.0.2)
+  set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${Boost_LIBRARIES})
+
+  # Build and run test program
+  include(CheckCXXSourceRuns)
+  check_cxx_source_runs("
+// CGAL test program
+
+#include <CGAL/Simple_cartesian.h>
+#include <CGAL/Point_3.h>
+#include <CGAL/Polyhedron_3.h>
+typedef CGAL::Simple_cartesian<double> SCK;
+typedef SCK::Point_3 Point;
+typedef CGAL::Polyhedron_3<SCK> Polyhedron_3;
+
+int main()
+{
+  // CGAL points
+  Point p1(0, 0, 0);
+  Point p2(1, 0, 0);
+  Point p3(0, 1, 0);
+  Point p4(0, 0, 1);
+
+  Polyhedron_3 P;
+  P.make_tetrahedron(p1, p2, p3, p4);
+
+  return 0;
+}
+" CGAL_TEST_RUNS)
```

Comment:
This fails on CentOS... Any idea why @jnbrunet ?

Reply:
Error is
```
FAILED: CMakeFiles/cmTC_151ec.dir/src.cxx.o
/opt/rh/llvm-toolset-7/root/usr/bin/clang++    -Wall -W -DCGAL_TEST_RUNS -o CMakeFiles/cmTC_151ec.dir/src.cxx.o -c src.cxx
In file included from src.cxx:4:
In file included from /usr/local/include/CGAL/Simple_cartesian.h:29:
In file included from /usr/local/include/CGAL/Cartesian/Cartesian_base.h:29:
In file included from /usr/local/include/CGAL/basic.h:30:
/usr/local/include/CGAL/config.h:122:10: fatal error: 'boost/config.hpp' file not found
#include <boost/config.hpp>
         ^~~~~~~~~~~~~~~~~~
1 error generated.
```

Reply:
After spending too much time on this, I disabled the build test with `-DCGAL_TEST_RUNS=TRUE`. 
Result: CGALPlugin build and tests are OK.

Reply:
Nevermind, I fixed this issue. No need to `-DCGAL_TEST_RUNS=TRUE` :+1: